### PR TITLE
used current aiogram for html escaping

### PR DESCRIPTION
**Pull request by emarc99**
Resolves issue #30 

Utilized the as_html method of the `Text` class to escape HTML symbols properly

Screenshot 2024-11-23 235441

### CODE REVIEW

File: tracker/telegram/bot.py [hunk 1]
```diff
@@ -118,10 +118,9 @@ def escape_html(text: str) -> str:
     Escapes HTML symbols in the text to ensure proper rendering in Telegram messages.
 
     :param text: The input string that may contain HTML symbols.
-    :return: A string with HTML symbols escaped, replacing '&' with '&amp;', '<' with '&lt;',
-             and '>' with '&gt;'.
+    :return: A string with HTML symbols escaped using the Text class.
     """
-    return text.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")
+    return Text(text).as_html()
```
Comment (iamnovichek) [inline tracker/telegram/bot.py]: Please use `html.unparse()` as described in issue

**Review by iamnovichek — CHANGES_REQUESTED** [review]
Your screenshot shouldn't show your code in your case but a working flow on Telegram. It could be a code sample like `code = "if  a>b: return a<<1;"` that should display properly on Telegram

File: tracker/telegram/bot.py (hunk 1 above)
Comment (emarc99) [reply tracker/telegram/bot.py]: I searched for html.unparse() in aiogram's recent official docs, is why I went with this route. Will make requested changes.

Comment (iamnovichek) [reply tracker/telegram/bot.py]: @emarc99 
```
from aiogram import html

code = "if  a>b: return a<<1;"
print(html.unparse(code))
```

Comment (emarc99) [reply tracker/telegram/bot.py]: Kindly check DM in Telegram.

File: tracker/telegram/bot.py [hunk 2]
```diff
@@ -118,9 +117,10 @@ def escape_html(text: str) -> str:
     Escapes HTML symbols in the text to ensure proper rendering in Telegram messages.
 
     :param text: The input string that may contain HTML symbols.
-    :return: A string with HTML symbols escaped using the Text class.
+    :return: A string with HTML symbols escaped, replacing '&' with '&amp;', '<' with '&lt;',
+             and '>' with '&gt;'.
     """
-    return Text(text).as_html()
+    return html.unparse()
```
Comment (iamnovichek) [inline tracker/telegram/bot.py]: You should pass the `text` param to `unparse` function

**Review by iamnovichek — CHANGES_REQUESTED** [review]

File: tracker/telegram/bot.py (hunk 2 above)
Comment (emarc99) [reply tracker/telegram/bot.py]: Done.

**Review by iamnovichek — APPROVED** [review]

**Comment by iamnovichek** [comment]
@emarc99 pls sync with main

**Pull request merged**